File: java/src/main/java/com/hydrapayments/sdk/exception/HydraException.java
package com.hydrapayments.sdk.exception;

public class HydraException extends RuntimeException {

    private final String errorCode;
    private final int statusCode;
    private final Object details;

    public HydraException(String message) {
        this(message, "API_ERROR", 0, null, null);
    }

    public HydraException(String message, String errorCode) {
        this(message, errorCode, 0, null, null);
    }

    public HydraException(String message, String errorCode, int statusCode, Object details) {
        this(message, errorCode, statusCode, details, null);
    }

    public HydraException(String message, String errorCode, int statusCode, Object details, Throwable cause) {
        super(message, cause);
        this.errorCode = errorCode;
        this.statusCode = statusCode;
        this.details = details;
    }

    public String getErrorCode() {
        return errorCode;
    }

    public int getStatusCode() {
        return statusCode;
    }

    public Object getDetails() {
        return details;
    }
}

File: dotnet/src/HydraPayments.Sdk/Exceptions/HydraException.cs
namespace HydraPayments.Sdk.Exceptions;

public class HydraException : Exception
{
    public string ErrorCode { get; }
    public int StatusCode { get; }
    public string? Details { get; }

    public HydraException(int statusCode, string errorCode, string message, string? details = null)
        : base(message)
    {
        StatusCode = statusCode;
        ErrorCode = errorCode;
        Details = details;
    }

    public HydraException(int statusCode, string message, string? details = null)
        : this(statusCode, "API_ERROR", message, details)
    {
    }
}
